### ISSUE TITLE
Fix duplicate running time entries in database

### ISSUE THREAD

**Issue by amulware** [toggl/mobileapp#2102]
It happens. It crashes the app. Uh oh.

Here are several stack traces that I _think_ are related.

https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/6047277ea69dfd9257b04797321b6c9f68abb357?appVersion=1.5
```
System.InvalidOperationException: Sequence contains more than one matching element

RealmResultsVisitor.VisitMethodCall (System.Linq.Expressions.MethodCallExpression node)
MethodCallExpression.Accept (System.Linq.Expressions.ExpressionVisitor visitor)
ExpressionVisitor.Visit (System.Linq.Expressions.Expression node)
RealmResultsProvider.Execute (System.Linq.Expressions.Expression expression)
RealmResultsProvider.Execute[T] (System.Linq.Expressions.Expression expression)
Queryable.SingleOrDefault[TSource] (System.Linq.IQueryable`1[T] source, System.Linq.Expressions.Expression`1[TDelegate] predicate)
RealmAdapter`2[TRealmEntity,TModel].resolvePotentialRivals (Realms.Realm realm, TRealmEntity entity, Toggl.PrimeRadiant.IRivalsResolver`1[TModel] resolver, System.Collections.Generic.List`1[T] results)
RealmAdapter`2[TRealmEntity,TModel].BatchUpdate (System.Collections.Generic.IEnumerable`1[T] batch, System.Func`3[T1,T2,TResult] conflictResolution, Toggl.PrimeRadiant.IRivalsResolver`1[TModel] rivalsResolver)
Repository`1+<>c__DisplayClass2_0[TModel].<BatchUpdate>b__0 ()
BaseStorage`1+<>c__DisplayClass8_0`1[TModel,T].<CreateObservable>b__0 (System.IObserver`1[T] observer)
```

https://appcenter.ms/orgs/Toggl/apps/Toggl-iOS/crashes/groups/0c5756bf4d4062b89886e54e0aeb2476033088ee?appVersion=1.5
```
System.InvalidOperationException: Sequence contains more than one element

Enumerable.SingleOrDefault[TSource] (System.Collections.Generic.IEnumerable`1[T] source)
TimeEntriesDataSource+<>c.<Stop>b__29_1 (System.Collections.Generic.IEnumerable`1[T] timeEntries)
Select`2+_[TSource,TResult].OnNext (TSource value)
ExceptionDispatchInfo.Throw ()
DefaultExceptionServices.Rethrow (System.Exception exception)
ExceptionHelpers.ThrowIfNotNull (System.Exception exception)
```

These occur in 1.5 and account for a good 40% of crashes.

A thought we had on slack just now: It might be finding running entries with IsDeleted == true and get confused by those.

**Comment by simonrozsival** [comment]
The thing is: these stack traces don't tell us how it is possible to get to a state with two running TEs, it just tells us that once there are two running TEs, it crashes (it is designed to do that).

We must find a reproducible way of getting two running TEs in the app.

**Comment by simonrozsival** [comment]
I am not sure how to debug this issue - I tried several times and tried to find some weird combination of data for which the conflict resolution fails - my best guess is addressed in PR #2378.

I think unless we have a way how to reproduce it, we should either
a) Handle the situation when there are two (or more?) running TEs in the database gracefully (apply conflict resolution again).
b) Try to catch the exception ourselves and track the conditions under which the inconsistency originates - is it push sync, pull sync, or local updates (starting a new TE)? This is something that we are lacking in the trace which we have in AppCenter/

**Comment by amulware** [comment]
I'd like to do b) so we can maybe fix the problem properly, instead of just working around it.

**Comment by amulware** [comment]
However, what about my comment on hitting the continue button a lot? Can that be reproduced in the simulator? If so, that should provide for a way to debug, no?

**Comment by simonrozsival** [comment]
This was fixed in #2573.
